applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -75,60 +79,87 @@ public static Builder builder() {
-
+        ConfigurationSetting result = validateSetting(setting, true);
         return service.setKey(serviceEndpoint, result.key(), result.label(), result, null, getETagValue(ETAG_ANY));
     }
 
     /**
-     * Creates or updates a configuration value in the service.
+     * Creates or updates a configuration value in the service with the given key. Partial updates are not supported.
+     *
+     * @param key The key for the configuration setting to create or update.
+     * @param value The value of this configuration setting.
+     * @return ConfigurationSetting that was created or updated.
+     * @throws IllegalArgumentException If {@code key} or {@code value} are {@code null} or an empty string.
+     */
+    public Mono<Response<ConfigurationSetting>> setSetting(String key, String value) {
+        return setSetting(new ConfigurationSetting().key(key).value(value));
+    }
+
+    /**
+     * Creates or updates a configuration value in the service. Partial updates are not supported.
      *
      * <p>
-     * If {@link ConfigurationSetting#etag()} is specified, the configuration value is updated if the current setting's
+     * If {@link ConfigurationSetting#etag() etag} is specified, the configuration value is updated if the current setting's
      * etag matches. If the etag's value is equal to {@link ConfigurationClient#ETAG_ANY}, the setting will always be
      * updated.
      *
-     * <p>
-     * The label value for the ConfigurationSetting is optional.
-     *
      * @param setting The configuration setting to create or update.
      * @return ConfigurationSetting that was created or updated.
      * @throws NullPointerException If {@code setting} is {@code null}.
-     * @throws IllegalArgumentException If {@link ConfigurationSetting#key()} is {@code null} or an empty string.
-     * @throws ServiceRequestException If the {@link ConfigurationSetting#etag()} was specified, is not
-     * {@link ConfigurationClient#ETAG_ANY}, and the current configuration
-     * value's etag does not match.
+     * @throws IllegalArgumentException If {@link ConfigurationSetting#key() key} or {@link ConfigurationSetting#value() value}
+     * are {@code null} or an empty string.
+     * @throws ServiceRequestException If the {@link ConfigurationSetting#etag() etag} was specified, is not
+     * {@link ConfigurationClient#ETAG_ANY}, and the current configuration value's etag does not match.
      */
     public Mono<Response<ConfigurationSetting>> setSetting(ConfigurationSetting setting) {
-        ConfigurationSetting result = validateSetting(setting);
+        ConfigurationSetting result = validateSetting(setting, true);
 
         return service.setKey(serviceEndpoint, result.key(), result.label(), result, getETagValue(result.etag()), null);
     }
 
+    /**
+     * Updates an existing configuration value in the service with the given key. The setting must already exist.
+     *
+     * <p>
+     * Partial updates are not supported.
```

Comment:
Weird JavaDoc for two reasons: missing a closing </p> element (more recent JavaDoc versions enforce HTML5 compliance), but also, it's weird to have a one-sentence paragraph. I would just add this to the end of the previous paragraph?

Reply:
![image](https://user-images.githubusercontent.com/10136526/55753461-39a29680-59ff-11e9-9948-48e61efe463b.png)
Odd is that when I added the `</p>` and ran the javadoc commands, I get this error.

Reply:
odd. I wonder if it would be fixed if you put the closing paragraph at the end of the same line, rather than on a new line?

Reply:
`</p>` is not necessary when it is followed by `<ol>` and shows up as an error in the repo (on that mozilla link below) when I paste my javadoc in there. There are some other situations that don't require a `</p>`. I learned a lot.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/p